
http://gcc.gnu.org/bugzilla/show_bug.cgi?id=46179

/tmp/cceVtxzz.s: Assembler messages:
/tmp/cceVtxzz.s:709: Error: syntax error -- statement `move.l %d1,last.5137@TLSLE+4(%a0)' ignored
/tmp/cceVtxzz.s:886: Error: syntax error -- statement `move.l %d1,last.5137@TLSLE+4(%a0)' ignored


diff -pru gcc-4.5-20101021.orig/gcc/config/m68k/m68k.c gcc-4.5-20101021/gcc/config/m68k/m68k.c
--- gcc-4.5-20101021.orig/gcc/config/m68k/m68k.c	2010-06-15 06:01:39.000000000 +1000
+++ gcc-4.5-20101021/gcc/config/m68k/m68k.c	2010-10-29 02:04:44.000000000 +1100
@@ -3478,6 +3478,7 @@ handle_move_double (rtx operands[2],
 
   /* Normal case: do the two words, low-numbered first.  */
 
+  m68k_final_prescan_insn (NULL, operands, 2);
   handle_movsi (operands);
 
   /* Do the middle one of the three words for long double */
@@ -3488,6 +3489,7 @@ handle_move_double (rtx operands[2],
       if (addreg1)
 	handle_reg_adjust (addreg1, 4);
 
+      m68k_final_prescan_insn (NULL, middlehalf, 2);
       handle_movsi (middlehalf);
     }
 
@@ -3498,6 +3500,7 @@ handle_move_double (rtx operands[2],
     handle_reg_adjust (addreg1, 4);
 
   /* Do that word.  */
+  m68k_final_prescan_insn (NULL, latehalf, 2);
   handle_movsi (latehalf);
 
   /* Undo the adds we just did.  */
